src/cc65/goto.c

```diff
@@ -54,21 +62,102 @@ void GotoStatement (void)
     NextToken ();
 
     /* Label name must follow */
-    if (CurTok.Tok != TOK_IDENT) {
-
-        Error ("Label name expected");
-
-    } else {
+    if (CurTok.Tok == TOK_IDENT) {
 
         /* Add a new label symbol if we don't have one until now */
         SymEntry* Entry = AddLabelSym (CurTok.Ident, SC_REF | SC_GOTO);
 
         /* Jump to the label */
         g_jump (Entry->V.L.Label);
-    }
 
-    /* Eat the label name */
-    NextToken ();
+        /* Eat the label name */
+        NextToken ();
+
+    } else if (CurTok.Tok == TOK_STAR && IS_Get (&Standard) >= STD_CC65) {
+        SymEntry *arr, *idx, *cur;
+        SymTable *tab;
+        ExprDesc desc;
+        CodeEntry *E;
+        unsigned char val;
+        unsigned I;
+
+        NextToken ();
+
+        /* arr[foo], we only support simple foo for now */
+        if (CurTok.Tok == TOK_IDENT &&
+            (arr = FindSym (CurTok.Ident))) {
+            NextToken ();
+
+            /* Find array size */
+            if (!IsTypeArray (arr->Type) || SizeOf (arr->Type) == 0 ||
+                SizeOf (GetElementType(arr->Type)) != 2) {
+                Error ("Expected array");
```

Comment:
The current implementation requires static label arrays.  You should add a test for the `SC_STATIC` flag.  And, make the error message say, `"Expected a static array"`.